tests/unittests/CoreText/CTFrameTests.mm

```diff
@@ -0,0 +1,78 @@
+
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include <TestFramework.h>
+#include "Starboard.h"
+#import <UIKit/UIKit.h>
+#import <Foundation/Foundation.h>
+#import <CoreText/CoreText.h>
+#import <CoreFoundation/CFAttributedString.h>
+
+static NSAttributedString* getAttributedString(NSString* str) {
+    UIFontDescriptor* fontDescriptor = [UIFontDescriptor fontDescriptorWithName:@"Times New Roman" size:40];
+    UIFont* font = [UIFont fontWithDescriptor:fontDescriptor size:40];
+
+    NSRange wholeRange = NSMakeRange(0, str.length);
+    NSMutableAttributedString* string = [[[NSMutableAttributedString alloc] initWithString:str] autorelease];
+    [string addAttribute:NSForegroundColorAttributeName value:[UIColor redColor] range:wholeRange];
```

Comment:
fyi, there is a convenience initializer, [NSAttributedString initWithString:attributes:] that iirc applies the attributes to the whole range.